src/app_service/service/keycard/service.nim

```diff
@@ -170,8 +175,12 @@ QtObject:
       debug "keycardResumeFlow", kcServiceCurrFlow=($self.currentFlow), payload=payload, response=response
 
   proc cancelCurrentFlow*(self: Service) =
+    # Do not remove the sleep 700
+    # This sleep prevents a crash on intel MacOS
+    # with errors like bad flushGen 12 in prepareForSweep; sweepgen 0
+    if status_const.IS_MACOS and status_const.IS_INTEL:
+      sleep 700
     let response = keycard_go.keycardCancelFlow()
-    sleep(200)
```

Comment:
@siddarthkay we need this after calling cancel flow from the keycard lib, always.

Reply:
Ah thanks for taking a look! I may have messed this up during rebase. I’ll add it in a bit

Reply:
@saledjenic : done

Reply:
thanks